linker/descriptors.go

```diff
@@ -1441,62 +1499,62 @@ func (f *fldDescriptor) DefaultEnumValue() protoreflect.EnumValueDescriptor {
 }
 
 func (f *fldDescriptor) ContainingOneof() protoreflect.OneofDescriptor {
-	if f.IsExtension() {
-		return nil
-	}
-	if f.proto.OneofIndex == nil {
-		return nil
-	}
-	parent := f.parent.(*msgDescriptor)
-	index := int(f.proto.GetOneofIndex())
-	ood := parent.proto.OneofDecl[index]
-	fqn := parent.fqn + "." + ood.GetName()
-	return f.file.asOneOfDescriptor(ood, f.file, parent, index, fqn)
+	return f.oneof
 }
 
 func (f *fldDescriptor) ContainingMessage() protoreflect.MessageDescriptor {
-	if !f.IsExtension() {
-		return f.parent.(*msgDescriptor)
+	if f.extendee != nil {
+		return f.extendee
 	}
-	return f.file.ResolveMessageType(protoreflect.FullName(f.proto.GetExtendee()))
+	return f.parent.(protoreflect.MessageDescriptor)
 }
 
 func (f *fldDescriptor) Enum() protoreflect.EnumDescriptor {
 	if f.proto.GetType() != descriptorpb.FieldDescriptorProto_TYPE_ENUM {
 		return nil
 	}
-	return f.file.ResolveEnumType(protoreflect.FullName(f.proto.GetTypeName()))
+	if f.enumType == nil {
+		f.enumType = f.file.ResolveEnumType(protoreflect.FullName(f.proto.GetTypeName()))
```

Comment:
Is there a reason we don't initialize fields like this up front as well?

Reply:
Doh! We _do_ initialize them up front. I just forgot to update these methods (here, below in `Message()`, and further below for method's input and output types). Fixed.